packages/core/src/components/tabs/_tabs.scss

```diff
@@ -103,6 +111,10 @@ $tab-indicator-width: 3px !default;
   position: relative;
   vertical-align: top;
 
+  display: flex;
```

Comment:
This is the dangerous bit. If the tab is used with unwrapped content, flex will apply to it. It will align content on a row and vertically centred.
It removes the need for using `line-height` making the `fill` implementation much nicer and will allow for adding `icon` and `tag` prop in [this pr](https://github.com/palantir/blueprint/pull/5950)